scripts/cmake/vcpkg_configure_cmake.cmake

```
@@ -211,6 +211,12 @@ function(vcpkg_configure_cmake)
         endif()
     endif()
 
+    # make sure to escape multiple architectures in VCPKG_TARGET_ARCHITECTURE
+    string(REPLACE ";" "\\;" _VCPKG_TARGET_ARCHITECTURE "${VCPKG_TARGET_ARCHITECTURE}")
```

And then here:

```suggestion
    string(REPLACE ";" "\\;" _VCPKG_TARGET_ARCHITECTURE "${VCPKG_TARGET_ARCHITECTURE}")
    string(REPLACE "x64" "x86_64" _VCPKG_TARGET_ARCHITECTURE "${_VCPKG_TARGET_ARCHITECTURE}")
```

thus, we actually write the vcpkg names for these targets, as opposed to the cmake names. I'm not sure if this is the right way, but it makes sense to me... cc @ras0219-msft ?

---

The reason I think this is that in `x86-windows`, we set `VCPKG_TARGET_ARCHITECTURE` to `x86`, not `Win32`.

---

Also, this _should_ break x64-osx.